tests/lib/eslint/eslint.js

```diff
@@ -5100,73 +5100,155 @@ describe("ESLint", () => {
+                    fix: true,
+                    fixTypes: ["suggestion"]
+                });
+                const inputPath = getFixturePath("fix-types/fix-only-prefer-arrow-callback.js");
+                const outputPath = getFixturePath("fix-types/fix-only-prefer-arrow-callback.expected.js");
+                const results = await eslint.lintFiles([inputPath]);
+                const expectedOutput = fs.readFileSync(outputPath, "utf8");
 
-        it("should not fix non-style rules when fixTypes has only 'layout'", async () => {
-            eslint = new ESLint({
-                cwd: path.join(fixtureDir, ".."),
-                overrideConfigFile: true,
-                fix: true,
-                fixTypes: ["layout"]
+                assert.strictEqual(results[0].output, expectedOutput);
             });
-            const inputPath = getFixturePath("fix-types/fix-only-semi.js");
-            const outputPath = getFixturePath("fix-types/fix-only-semi.expected.js");
-            const results = await eslint.lintFiles([inputPath]);
-            const expectedOutput = fs.readFileSync(outputPath, "utf8");
 
-            assert.strictEqual(results[0].output, expectedOutput);
+            it("should fix both style and problem rules when fixTypes has 'suggestion' and 'layout'", async () => {
+                eslint = new ESLint({
+                    cwd: path.join(fixtureDir, ".."),
+                    overrideConfigFile: true,
+                    fix: true,
+                    fixTypes: ["suggestion", "layout"]
+                });
+                const inputPath = getFixturePath("fix-types/fix-both-semi-and-prefer-arrow-callback.js");
+                const outputPath = getFixturePath("fix-types/fix-both-semi-and-prefer-arrow-callback.expected.js");
+                const results = await eslint.lintFiles([inputPath]);
+                const expectedOutput = fs.readFileSync(outputPath, "utf8");
+
+                assert.strictEqual(results[0].output, expectedOutput);
+            });
         });
 
-        it("should not fix style or problem rules when fixTypes has only 'suggestion'", async () => {
-            eslint = new ESLint({
-                cwd: path.join(fixtureDir, ".."),
-                overrideConfigFile: true,
-                fix: true,
-                fixTypes: ["suggestion"]
+        describe("with lintText", () => {
+            it("should throw an error when pattern is passed instead of code", () => {
+                assert.rejects(async () => {
+                    eslint = new ESLint({
+                        cwd: path.join(fixtureDir, ".."),
+                        overrideConfigFile: true,
+                        fix: true,
+                        fixTypes: ["layout"]
+                    });
+                    const inputPath = getFixturePath("fix-types/fix-only-semi.js");
+
+                    await eslint.lintText([inputPath]);
+                }, /'code' must be a string\./iu);
             });
```

Comment:
This is an interesting test case, but my feeling is that it does not belong in this test suite (you'd get the same error without `fix` or `fixTypes`). A similar, but more minimal test case with a non-string argument is already included in the "lintText()" suite:

https://github.com/eslint/eslint/blob/41d02066935b987d2e3b13a08680cc74d7067986/tests/lib/eslint/eslint.js#L905-L908

Reply:
I will remove this test.